src/PrestaShopBundle/Routing/LegacyUrlConverter.php

```diff
@@ -0,0 +1,268 @@
+        $this->controllersActions[$controller][$action] = $routeName;
+    }
+
+    /**
+     * @param array $parameters
+     *
+     * @return string
+     *
+     * @throws ArgumentException
+     * @throws RouteNotFoundException
+     */
+    public function convertByParameters(array $parameters)
+    {
+        if (empty($parameters['controller'])) {
+            throw new ArgumentException('Missing required controller argument');
+        }
+
+        if (!isset($this->controllersActions[$parameters['controller']])) {
+            throw new RouteNotFoundException(
+                'Could not find a route matching for legacy controller: %s',
+                [$parameters['controller']]
+            );
+        }
+
+        $routeName = $this->findRouteNameByParameters($parameters);
+        $parameters = $this->convertParametersForRoute($parameters, $routeName);
+
+        return $this->router->generate($routeName, $parameters, UrlGeneratorInterface::ABSOLUTE_URL);
+    }
+
+    /**
+     * @param string $url
+     *
+     * @return string
+     *
+     * @throws ArgumentException
+     * @throws RouteNotFoundException
+     */
+    public function convertByUrl($url)
+    {
+        $parsedUrl = parse_url($url);
+        $parameters = array();
+        if (isset($parsedUrl['query'])) {
+            parse_str($parsedUrl['query'], $parameters);
+        }
+
+        return $this->convertByParameters($parameters);
+    }
+
+    /**
+     * This conversion method is used by the listener, indeed it needs to update the
+     * router context because it is executed before the RouterListener. Thus the router
+     * does not have the appropriate information to generate an admin url and would redirect
+     * to the front office.
+     *
+     * @param Request $request
+     *
+     * @return string
+     *
+     * @throws ArgumentException
+     * @throws RouteNotFoundException
+     */
+    public function convertByRequest(Request $request)
+    {
+        $this->router->getContext()->fromRequest($request);
+
+        return $this->convertByUrl($request->getUri());
+    }
+
+    /**
+     * @param array $parameters
+     * @param string $routeName
+     *
+     * @return array
+     */
+    private function convertParametersForRoute(array $parameters, $routeName)
+    {
+        $route = $this->legacyRoutes[$routeName];
+        $routeDefaults = $route->getDefaults();
+        if (!empty($routeDefaults['_legacy_parameters']) && is_array($routeDefaults['_legacy_parameters'])) {
+            foreach ($routeDefaults['_legacy_parameters'] as $legacyParameter => $parameter) {
+                if (isset($parameters[$legacyParameter])) {
+                    $parameters[$parameter] = $parameters[$legacyParameter];
+                    unset($parameters[$legacyParameter]);
+                }
+            }
+        }
+
+        $legacyAction = $this->getActionFromParameters($parameters);
+        unset($parameters['controller']);
```

Comment:
make sure key exist before unset?

Reply:
That's not needed, it won't throw any notice.

Reply:
yep, unset the world! don't give a sh**